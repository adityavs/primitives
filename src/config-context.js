import React, { Component } from 'react';

import defaultConfig from './config';
import { merge } from './utils/object';

const ConfigContext = React.createContext();
const ConfigConsumer = ConfigContext.Consumer;

export default class ConfigProvider extends Component {
  constructor(props) {
    super(props);
    this.updateConfig = this.updateConfig.bind(this);

    this.state = {
      config: defaultConfig
    };
  }

  updateConfig(newConfig = {}) {
    this.setState((state) => {
      const newState = {
        ...state,
        config: merge(state.config, newConfig)
      };
      return newState;
    });
  }

  render() {
    const { children } = this.props;
    const { config } = this.state;

    const { colors, scheme } = config;

    const value = {
      config: {
        ...config,
        colorScheme: colors[scheme]
      },
      updateConfig: this.updateConfig
    };

    return (
      <ConfigContext.Provider value={value}>{children}</ConfigContext.Provider>
    );
  }
}

export { ConfigProvider, ConfigConsumer };
