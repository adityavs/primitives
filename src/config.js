const weights = {
  extralight: '100',
  thin: '200',
  light: '300',
  normal: '400',
  medium: '500',
  semibold: '600',
  bold: '700',
  heavy: '800',
  black: '900'
};

const defaultTheme = {
  fontSize: [32, 24, 20, 16, 14, 12, 10, 8],
  fontWeight: 'normal',
  fontFamily: undefined,
  weights,
  spacer: [26, 22, 18, 14, 10, 6],
  colors: {
    brand: '#28D9AA',
    background: '#FFF',
    text: '#000'
  },
  isDark: false
}

const darkTheme = {
  ...defaultTheme,
  colors: {
    brand: '#28D9AA',
    background: '#000',
    text: '#FFF'
  },
  isDark: true
}

export default {
  activeTheme: 'default',
  default: defaultTheme,
  dark: darkTheme
}
