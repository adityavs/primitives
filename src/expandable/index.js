import Expandable from './expandable';

export { Expandable };
