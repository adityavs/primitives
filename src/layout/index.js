import BaseLayout from './base'
import Column from './column'
import Row from './row'
import Float from './float'

export {
  BaseLayout,
  Column,
  Row,
  Float
}
