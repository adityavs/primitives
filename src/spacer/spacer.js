import React, { useContext } from 'react';
import { View } from 'react-native';

import ConfigContext from '../config-context';
import { makeStyle } from '../utils/style';

const getDimensions = (size) => {
  return {
    width: size,
    height: size
  };
};

const Spacer = (props) => {
  const { size = 'm', fullWidth, ...rest } = props;
  const { theme } = useContext(ConfigContext);
  const { spacer } = theme;

  const baseStyle = makeStyle([
    getDimensions(spacer[size]),
    fullWidth && { width: '100%' },
    { ...rest }
  ]);

  return <View style={baseStyle} />;
};

export default Spacer;
