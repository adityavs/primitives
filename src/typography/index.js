import React, { useContext } from 'react';

import BaseText from './base';
import ConfigContext from '../config-context';
import config from '../config';

const makeHeadingComponent = ({ key }) => {
  return ({ children, ...rest }) => {
    const { theme } = useContext(ConfigContext);
    const { fontSize } = theme;

    return (
      <BaseText fontSize={fontSize[key]} {...rest}>
        {children}
      </BaseText>
    );
  };
};

const [H1, H2, H3, H4, H5, H6, H7, H8] = Object.entries(
  config.default.fontSize
).map(([key]) => makeHeadingComponent({ key }));

export { BaseText, H1, H2, H3, H4, H5, H6, H7, H8 };
