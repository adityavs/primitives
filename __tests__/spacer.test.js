/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import { ConfigProvider } from '../src/config-context';
import { Spacer } from '../src/spacer';

describe('<Spacer>', () => {
  it('should render spacer - default size', () => {
    const tree = renderer
      .create(
        <ConfigProvider>
          <Spacer />
        </ConfigProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
