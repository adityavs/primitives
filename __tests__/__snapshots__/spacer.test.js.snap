// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spacer> should render spacer - default size 1`] = `
<View
  style={
    Object {
      "height": 18,
      "width": 18,
    }
  }
/>
`;
